src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -202,6 +202,12 @@ private static Assembly ResolveForUnknownRequestor(string requestedAssemblyName)
                     foreach (string loadedAssemblyFullPath in s_assembliesFromFiles.Keys)
                     {
                         string directoryPath = Path.GetDirectoryName(loadedAssemblyFullPath);
+
+                        if (!string.IsNullOrEmpty(requestedAssemblyIdentity.CultureName))
+                        {
+                            directoryPath = Path.Combine(directoryPath, requestedAssemblyIdentity.CultureName);
```

Comment:
Additionally, is it also worth looking for sub-folder with name: "CultureInfo.GetCultureInfo(requestedAssemblyIdentity.CultureName)?.LCID"?